### PR TITLE
Clear command

### LINKED ISSUE #137: Implement clear validation

### PR DESCRIPTION
**Pull request by JohannsenLum**
This PR includes a validation added for clear command and new methods isValidConfirm and ClearCommandParser

Added test cases to test the new added functions and also updated the User Guide

Why this is needed:
- Ensures that the user do not accidentally delete the whole reservation list which is IRREVERSIBLE 

Fixes #137

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -191,7 +191,11 @@ Examples:
 
 Clears all entries from the ReserveMate
 
-Format: `clear`
+Format: `clear cfm`
+
+* A confirmation flag **'cfm' is required** to confirm the clear action
+    * If cfm is not included, the clear action will not occur.
+    * keyword 'cfm' is case-sensitive
```
Comment (wowwwp) [inline docs/UserGuide.md:198]: Good job on making it clear it is case sensitive

File: src/test/java/seedu/reserve/logic/commands/ClearCommandTest.java [hunk 2]
```diff
@@ -26,7 +33,83 @@ public void execute_nonEmptyReserveMate_success() {
...
+    @Test
+    public void isValidConfirm_validConfirmation_returnsTrue() throws ParseException {
+        // Single valid confirmation "cfm"
+        assertTrue(ClearCommand.isValidConfirm("cfm"));
+
+        // Single valid confirmation with extra spaces
+        assertTrue(ClearCommand.isValidConfirm("cfm  "));
+
+        // Any other word than cfm is still valid but will set isConfimed = false
+        assertTrue(ClearCommand.isValidConfirm("false"));
```
Comment (wowwwp) [inline src/test/java/seedu/reserve/logic/commands/ClearCommandTest.java:48]: Good job on checking the edge cases, perhaps you can add isValidConfirm("")

**Review by wowwwp — COMMENTED** [review]
lgtm, just a few minor issues make sure to resolve conflicts before merging!

File: src/test/java/seedu/reserve/logic/commands/ClearCommandTest.java (hunk 2 above)
Comment (JohannsenLum) [reply src/test/java/seedu/reserve/logic/commands/ClearCommandTest.java:48]: will add more tests! next week
